Automattic/woocommerce-payments
# Fix error message when adding new credit card

When a customer adds a new credit card inside My account > Payment Methods using other gateways than WC Payments the error "Your credit card number is invalid" would pop up but the credit card would be added anyways as the error was not related to the payment gateway selected.

Fixes #1149 

#### Changes proposed in this Pull Request
When the customer clicked on Add payment method, WC Payments validated if the information was correctly filled on WC Payments fields, even when the customer selected other Payment Gateway. I've added a conditional to make sure we only validate WC Payments fields if the customer selected a WC Payments Gateway.

Also added tests to guarantee this problem does not happen again in the future.

#### Testing instructions

- Have WC Payments installed and configured
- Have some other Credit Card Payment method installed and configured
- Log in as a customer
- Go to My account > Payment Methods
- Click on Add Payment Method
- Fill the **other gateway** Credit Card details
- Try to submit
- The error "Your card number is incomplete" should not show up

-------------------


**Post merge**
